### ISSUE TITLE
Create a monthly invoicing process for personnel and cloud costs

### ISSUE THREAD

**Issue by choldgraf** [2i2c-org/team-compass#355]
### Context

We currently define [our invoicing process here](https://team-compass.2i2c.org/en/latest/projects/managed-hubs/sales.html). In brief, we must manually send an email to every community that we wish to invoice every month. When they confirm that the service has been delivered, we send that to CS&S and they can send an invoice.

By having this manual step each month, we create more manual toil that we need to manage and carry out. This makes our invoicing process more complicated than it needs to be, given that most communities likely just want a monthly invoice (if we expect repeating business), not a confirmation email first.

**Proposal**

We should either:

- Work with CS&S to define a process of automatically sending invoices for communities that expect it
- OR: create a more semi-automated process for us to send out invoices to the communities we must bill each month, so there is less manual work

This would reduce the number of manual steps associated with billing/sales, which would make the service more scalable and less prone to error.

### Implementation guide and constraints

We'll likely need to collaborate directly with CS&S on this, to see if they are legally restricted by their accounting practices in someway.

### Updates and ongoing work

- [x] Finalize an alpha draft of the [price quote document](https://docs.google.com/document/d/11Hhhprw7_11qm9chdZ6C22ei9VQkX_Bze8_1TUJ-Bhg/edit#)
- [x] Send that document to a few communities and get them to agree to a specific price / set of services.
- [x] When they approve, forward to CS&S and have them issue some invoices
- [ ] https://github.com/2i2c-org/team-compass/issues/519
    - Related engineering goal: https://github.com/2i2c-org/infrastructure/issues/1853
- [x] https://github.com/2i2c-org/team-compass/issues/555
- [x] #527
- [ ] Write up this process in our team compass

**Comment by choldgraf** [comment]
### Update: converging on a few documents

@colliand and I met with the CS&S team today to go through the invoicing process that we are shooting for. Here are a few highlights:

- [This price quote document](https://docs.google.com/document/d/11Hhhprw7_11qm9chdZ6C22ei9VQkX_Bze8_1TUJ-Bhg/edit#) will be the final thing that 2i2c gets communities to sign, before we hand off to CS&S
- That document should be all that CS&S needs to start sending invoices each month.
- For communities where we are paying their cloud costs, we need to be able to generate a **monthly table of cloud costs, broken down by community**.
- Our next steps are to:
  - Finalize the price quote document
  - Build rough infrastructure to generate monthly "cloud costs reports" for each community
  - Prototype this process with a few communities

**Comment by choldgraf** [comment]
I've added an extra task here to finish a one-time recoup of our cloud costs from 2022. I've also included a bunch of data about what our actual costs were, and which projects they belong to:

- https://github.com/2i2c-org/team-compass/issues/555

**Comment by choldgraf** [comment]
Note that the Engineering team is setting **Billing as a Q4 priority**, but they need guidance from this team about what needs implementation. See their tracking issue below and please provide feedback if you can:

- https://github.com/2i2c-org/infrastructure/issues/1853

**Comment by damianavila** [comment]
We are having a dedicated meeting with @jmunroe and @yuvipanda later today to get into the specifics of what would be needed and how to accomplish it.

**Comment by choldgraf** [comment]
I'm pasting in an update that @yuvipanda gave in Slack that I think was regarding this meeting:

> Action items from it are:
> 1. [@Damián Avila](https://2i2c.slack.com/team/U01GKCV6PBL) creates placeholder issues laying out the structure of planned work (for both backlog and future work),
> 2. I expand them over the course of this sprint,
> 3. We work on them from next sprint wherever possible,
> 4. [@James Munroe](https://2i2c.slack.com/team/U03JY6QHQTY) has all the info he needs currently for working on the backlog manually, except
> 5. [@yuvipanda](https://2i2c.slack.com/team/UKNHAAGHF) to ensure he has access to AWS billing info

**Comment by damianavila** [comment]
> Can somebody confirm who is the one responsible for driving this effort forward, and re-assign this issue to that person so that we have clarity?

There are several pieces from the summary you pasted above that will be run in parallel... and those tasks are already assigned to people. This is shared work so it is not clear to me if there is just one person to be assigned. Temporarily I have assigned this to myself as a representative of the group pushing this forward.

**Comment by choldgraf** [comment]
> This is shared work so it is not clear to me if there is just one person to be assigned. Temporarily I have assigned this to myself as a representative of the group pushing this forward.

Agreed that we need to divide and conquer the individual tasks, but we need to have one person responsible for the effort as a whole, and making sure we are making progress in all of these individual places. E.g.I think if we have a "tracking issue" that has lots of sub-issues, the person assigned to the tracking issue should be the one coordinating and delegating.

**Comment by damianavila** [comment]
In general, I agree with this point of view! Although I think this idea needs an iteration (most likely in the future, not now) where there are groups of people to be assigned instead of one person and "the group" is responsible for pushing things forward. But, GitHub is bad at the time to have grouped assignations AND we still need sole assignations given our size and current state of the organization.

**Comment by damianavila** [comment]
OK, I have updated https://github.com/2i2c-org/team-compass/issues/555#issuecomment-1316030732 with the next steps on @jmunroe's hands.

I also created placeholder issues inheriting from the [goal billing issue](https://github.com/2i2c-org/infrastructure/issues/1853) which @yuvipanda will further specify from the technical perspective.

**Comment by choldgraf** [comment]
### Closing this one as mostly completed

We've documented the process for invoicing ongoing contracts in a few PRs:

- #647 
- #698 

And we're tracking the improvements to cloud billing infrastructure here:

- https://github.com/2i2c-org/infrastructure/issues/2431

So I'll close this one as completed and we can focus our efforts on new issues that track iterations on this work
